pkgs/by-name/bi/bitwarden-directory-connector-cli/package.nix

```diff
@@ -0,0 +1,66 @@
+{
+  lib,
+  buildNpmPackage,
+  fetchFromGitHub,
+  jq,
+  python3,
+  pkg-config,
+  libsecret,
+  nodejs_18,
+}:
+buildNpmPackage rec {
+  pname = "bitwarden-directory-connector-cli";
+  version = "2023.10.0";
+  nodejs = nodejs_18;
+
+  src = fetchFromGitHub {
+    owner = "bitwarden";
+    repo = "directory-connector";
+    rev = "v${version}";
+    hash = "sha256-PlOtTh+rpTxAv8ajHBDHZuL7yeeLVpbAfKEDPQlejIg=";
+  };
+
+  postPatch = ''
+    ${lib.getExe jq} 'del(.scripts.preinstall)' package.json > package.json.tmp
+    mv -f package.json{.tmp,}
+  '';
```

Comment:
Please add `jq` to `nativeBuildInputs`.

Reply:
Read the previous discussion on the topic. This is a workaround for the fact that `postPatch` is carried through into the deps FOD, and `nativeBuildInputs` isn't.

Reply:
Then we should use `buildPackages.jq` to get binaries of the correct platform.

Reply:
Would I be correct in thinking that that would be importing jq into buildPackages only to export it as buildPackages.jq?

Reply:
Nope, `buildPackages` is just a toplevel nixpkgs attrset.

Reply:
So somethign like

```nix
  postPatch = ''
    ${lib.getExe buildPackages.jq} 'del(.scripts.preinstall)' package.json > package.json.tmp
    mv -f package.json{.tmp,}
  '';
```

And remove the jq from teh top of the file?

Reply:
Yes, and add buildPackages to the top.

Reply:
That works well, going to push in a moment